portfolio/src/main/webapp/js/script.js

```diff
@@ -11,19 +11,44 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+var mode = false;
 
 /**
- * Switches Light Mode of Page
+ * Sets Light mode of page based on cookies
  */
-function switchMode() {
-  var style = document.getElementById('pagestyle').getAttribute('href');
-  if (style == '') { 
+function setMode() { 
+  var value = getCookie('mode');
+  if(value=="dark"){
     switchSheet('css/nightmode.css');
     switchIcon('fas fa-sun');
-  } else { 
+    mode = true;
+  }else {
     switchSheet('');
     switchIcon('fas fa-moon');
+    mode = false;
+    }
+}
+
+/**
+ * @param {string} The string whose value you want to get from the cookie. 
+ */
+function getCookie(name) { 
+  var re = new RegExp(name + "=([^;]+)");
+  var value = re.exec(document.cookie);
+  return (value != null) ? unescape(value[1]) : null;
+}
+
+/**
+ * Switches Light Mode of Page
+ */
+function switchMode() {
+  if (!mode) { 
```

Comment:
Flip the if/else. Generally, the following structure is going to be clearer:

```
if (condition) {
} else {
}
```

instead of what you have:

```
if (!condition) {
} else {
}
```